src/v/cluster/partition.cc

```diff
@@ -902,6 +902,7 @@ partition::get_cloud_term_last_offset(model::term_id term) const {
 }
 
 ss::future<> partition::remove_persistent_state() {
+    _cloud_storage_probe = nullptr;
```

Comment:
Wondering if this belongs in `partition::stop()`? Also wondering if we should be calling `_cloud_storage_probe->clear_metrics()`, as we do with the `partition::_probe`?

Reply:
added clear metrics method